week4/homework/Program.cs

```diff
@@ -0,0 +1,123 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello this is homework 4 World!");
+/*
+A POST endpoint that calls POST users/add with a record with FirstName, 
+LastName and Age (this simulates creating a user)
+*/
+app.MapPost("/create", async (User userData) =>
+{
+    var httpClient = new HttpClient();
+    var response= await httpClient.PostAsJsonAsync("https://dummyjson.com/users/add", userData)
+        ;
+        System.Console.WriteLine(response);
+        var data = await response.Content.ReadFromJsonAsync<User>();
+        var responseData = new
+        {
+            StatusCode = response.StatusCode,
+            ReasonPhrase = response.ReasonPhrase,
+            data
+
+        };
+        return (responseData);
+});
+/*
+A POST endpoint that that calls Post products/add with a record with Title and 
+Price (this simulates creating a product)
+*/
+app.MapPost("/createProduct", async (Product productData) =>
+{
+    var httpClient = new HttpClient();
+    var response= await httpClient.PostAsJsonAsync("https://dummyjson.com/products/add", productData)
+        ;
+        System.Console.WriteLine(response);
+        var data = await response.Content.ReadFromJsonAsync<Product>();
+        var responseData = new

```

Comment:
If you want to improve this even more, you can create a new class or record for this so you avoid using the object
```
  var postResponce = new
  {
    StatusCode = postRequest.StatusCode,
    ReasonPhrase = postRequest.ReasonPhrase,
    data = UserDataResponce,
    Date = postRequest.Headers.Date,
  };
```

then you could have the Data set as a Generic type and then you can use the same class for all of the other responses.
Please reach out if you want to do this and need assistance

Reply:
Hi I made a new class but unable to find the datatype for those parameters. Could you please help me with this.

Reply:
Sorry for the late response. 
you can create a record:
```
public record ApiResponse<T>(HttpStatusCode StatusCode, string ReasonPhrase, T data);
```

then you use it like so
```
var responseData = new ApiResponse<User>(response.StatusCode, response.ReasonPhrase, data);
```